timesketch/frontend/src/components/Sigma/SigmaList.vue

```diff
@@ -127,14 +186,167 @@ export default {
       return this.$store.state.meta
     },
   },
+  created() {
+    if (this.timeline) {
+      ApiClient.getSketchTimelineAnalysis(this.sketch.id, this.timeline.id)
+        .then(response => {
+          this.analyses = response.data.objects[0]
+        })
+        .catch(e => { })
+    }
+    // If no timeline was specified then loop over all of them
+    else {
+      this.sketch.timelines.forEach(timeline => {
+        ApiClient.getSketchTimelineAnalysis(this.sketch.id, timeline.id)
+          .then(response => {
+            this.analyses = this.analyses.concat(response.data.objects[0])
+          })
+          .catch(e => { })
+      })
+    }
+  },
+  mounted() {
+    this.loadSketchSigmaTags()
+    this.loadSketchTTP()
+  },
   methods: {
-    parseSigma: function(event) {
-      ApiClient.getSigmaByText(this.text)
+    // Set debounce to 300ms if parseSigma is used.
+    parseSigma: _.debounce(function (rule_yaml) { // eslint-disable-line
+      this.parsing_issues = []
+      ApiClient.getSigmaRuleByText(rule_yaml)
         .then(response => {
           let SigmaRule = response.data.objects[0]
           this.parsed = SigmaRule
         })
-        .catch(e => {})
+        .catch(e => {
+          Snackbar.open({
+            message: 'Sigma rule parsing failed. See Browser console for more',
+            type: 'is-danger',
+            position: 'is-top',
+            indefinite: false,
+          })
+        })
+    }, 300),
+
+
+
+
+
+    addRule: function (event) {
+      if (this.save_button_text === "Create") {
+        ApiClient.getSigmaRuleByText(this.editingRule.rule_yaml)
+          .then(response => {
+            let SigmaRule = response.data.objects[0]
+            this.parsed = SigmaRule
```

Comment:
This is a repeat of what's happening in lines 255->259, maybe we can check for the state of save_button_text in the callback of this sucessful function?

Reply:
Actually, do we really need this call to `getSigmaRuleByText` in the first palce? isn't `createSigmaRule` also going to return an object that we can assign to `this.parsed`?

Reply:
I refactored that a bit, PTAL